File: test/arguments/without_grpc_code/python/WithoutGrpcCodeTest.py
import os
import pathlib

import unittest

from testutils import getZserioApi

class WithoutWriterCodeTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "without_grpc_code.zs", extraArgs=["-withoutGrpcCode"])

    def testCheckService(self):
        self.assertFalse(hasattr(self.api, "Service"))
        self.assertFalse(self._isFilePresent("Service.py"))

    def testCheckResponse(self):
        self.assertTrue(hasattr(self.api, "Response"))
        self.assertTrue(self._isFilePresent("Response.py"))

    def testCheckRequest(self):
        self.assertTrue(hasattr(self.api, "Request"))
        self.assertTrue(self._isFilePresent("Request.py"))

    def _isFilePresent(self, filename):
        fileFullPath = os.path.join(os.path.abspath(os.path.join(self.api.__file__, "..")), filename)
        file = pathlib.Path(fileFullPath)
        return file.is_file()

File: test/arguments/without_grpc_code/cpp/WithoutGrpcCodeTest.cpp
#include <fstream>
#include <string>

#include "gtest/gtest.h"

namespace without_grpc_code
{

class WithoutGrpcCode : public ::testing::Test
{
protected:
    bool isFilePresent(const char* fileName)
    {
        std::ifstream file(fileName);
        return file.is_open();
    }
};

TEST_F(WithoutGrpcCode, checkGrpcSerializationTraits)
{
    ASSERT_FALSE(isFilePresent(
            "arguments/without_grpc_code/gen/without_grpc_code/GrpcSerializationTraits.h"));
}

TEST_F(WithoutGrpcCode, checkService)
{
    ASSERT_FALSE(isFilePresent(
            "arguments/without_grpc_code/gen/without_grpc_code/Service.cpp"));
    ASSERT_FALSE(isFilePresent(
            "arguments/without_grpc_code/gen/without_grpc_code/Service.h"));
}

TEST_F(WithoutGrpcCode, checkResponse)
{
    ASSERT_TRUE(isFilePresent(
            "arguments/without_grpc_code/gen/without_grpc_code/Response.cpp"));
    ASSERT_TRUE(isFilePresent(
            "arguments/without_grpc_code/gen/without_grpc_code/Response.h"));
}

TEST_F(WithoutGrpcCode, checkRequest)
{
    ASSERT_TRUE(isFilePresent(
            "arguments/without_grpc_code/gen/without_grpc_code/Request.cpp"));
    ASSERT_TRUE(isFilePresent(
            "arguments/without_grpc_code/gen/without_grpc_code/Request.h"));
}

} // namespace without_writer_code
